### PR TITLE
fixed epsilon summand typo in AudioStreamPlayer3D::_get_attenuation_db

### PR DESCRIPTION
**Pull request by GlaDos28**
EDIT: read the second comment, too.

There is probably a typo in implementation of `AudioStreamPlayer3D::_get_attenuation_db`. Instead of 0.00001 in 2 of 3 cases there is 000001. According to the second case (`ATTENUATION_INVERSE_SQUARE_DISTANCE`), in first and third cases should be 0.00001, too.

Note the thing that in second case (`ATTENUATION_INVERSE_SQUARE_DISTANCE`) if distance is nearly zero, there is very large `att` (`1 / 0.00001` before `linear2db`). Perhaps, the 'typo' version with `1` is better in terms of `att` value, it is always `<= 1` before `linear2db`.

Nevertheless `000001` is a strange constant :)

### CODE REVIEW

**Comment by GlaDos28** [comment]
I think the right implementation wouble be with `1` in all three cases.

Consider first and second cases the same in terms of a problem.

1. When `p_distance = 0`, `1 / 0.00001 >> 0` and `att > 0`, what is wrong, i.e. the volume can not be more than original volume of 0.

With `'epsilon' = 1` there is `attr = linear2db(x), x <= 1` => `attr <= 0`, so the volume is always going down.

2, When `p_distance = 0`, `Math::log(0 + 0.00001)` gives some negative number `x` => `-20 * x > 0`, i.e. sound gets louder.

If `'epsilon' = 1` then `attr = -20 * Math::log(1) = 0`, so `0` is a maximal value, hence sound can no become louder than in the center of a source.

So, perhaps, we should write `1` in all three cases?

**Comment by reduz** [comment]
Well, actually, it DOES make sense that volume goes over 0db when an object is very close. This is why unit volume, size and max volume are specified. I find this more realistic than attenuating from 0db downwards.

In reality, decibels tend to go towards infinite the closer to the ear.

**Comment by reduz** [comment]
Maybe taking advantage of the new inspector, a graph could be plotted showing the attenuation curve, so you understand better why it was made this way.

File: scene/3d/audio_stream_player_3d.cpp [hunk 1]
```diff
@@ -206,15 +206,15 @@ float AudioStreamPlayer3D::_get_attenuation_db(float p_distance) const {
...
+			att = Math::linear2db(1.0 / ((p_distance / unit_size) + 1));
 		} break;
 		case ATTENUATION_INVERSE_SQUARE_DISTANCE: {
 			float d = (p_distance / unit_size);
 			d *= d;
-			att = Math::linear2db(1.0 / (d + 0.00001));
+			att = Math::linear2db(1.0 / (d + 1));
 		} break;
 		case ATTENUATION_LOGARITHMIC: {
-			att = -20 * Math::log(p_distance / unit_size + 000001);
```
Comment (reduz) [inline scene/3d/audio_stream_player_3d.cpp]: oh i see sorry, i meant to use 0.00001 here

File: scene/3d/audio_stream_player_3d.cpp [hunk 2]
```diff
@@ -206,15 +206,15 @@ float AudioStreamPlayer3D::_get_attenuation_db(float p_distance) const {
 	float att = 0;
 	switch (attenuation_model) {
 		case ATTENUATION_INVERSE_DISTANCE: {
-			att = Math::linear2db(1.0 / ((p_distance / unit_size) + 000001));
```
Comment (reduz) [inline scene/3d/audio_stream_player_3d.cpp]: Meant to use 0.00001 here, sorry

**Comment by aaronfranke** [comment]
These should all be replaced with the `CMP_EPSILON` keyword (which is equal to `0.00001`).

**Comment by GlaDos28** [comment]
Well, according to your words, my first version with `0.00001` was right. In a day I will fix it, using epsilon constant, as was suggested by aaronfranke.

**Pull request closed without merging**